icu4c/source/common/cmemory.h

```diff
@@ -725,9 +725,8 @@ class MemoryPool : public UMemory {
     }
 
     MemoryPool& operator=(MemoryPool&& other) U_NOEXCEPT {
-        fCount = other.fCount;
-        fPool = std::move(other.fPool);
-        other.fCount = 0;
+        std::swap(this->fCount, other.fCount);
```

Comment:
For consistency with the surrounding code, I suggest omitting `this->` here.

Reply:
Done.